actors/market/src/lib.rs

```diff
@@ -634,7 +640,13 @@ impl Actor {
                     .find_deal_state(rt.store(), id)?
                     // A deal with a proposal but no state is not activated, but then it should not be
                     // part of a sector that is terminating.
-                    .ok_or_else(|| actor_error!(illegal_argument, "no state for deal {}", id))?;
+                    .ok_or_else(|| {
+                        if id < st.next_id {
+                            ActorError::unchecked(EX_DEAL_EXPIRED, format!("deal {} expired", id))
+                        } else {
+                            ActorError::not_found(format!("no such deal {}", id))
+                        }
+                    })?;
```

Comment:
Revert this one. I think the change only makes obvious sense when looking up proposals, not deal states. This should revert changes to the sector termination tests, too, which showed the issue.

Reply:
Got it, Done 👍🏾